allennlp/notebooks/vocabulary.ipynb

```diff
@@ -0,0 +1,293 @@
+    "\n",
+    "A `Vocabulary` maps strings to integers, allowing for strings to be mapped to an\n",
+    " out-of-vocabulary token.\n",
+    "\n",
+    "Vocabularies can be fit to a particular dataset, which we use to decide which tokens are\n",
+    " in-vocabulary, or alternatively, they can be loaded directly from a static vocabulary file.\n",
+    "\n",
+    "\n",
+    "First, let's import the vocabulary class from `allennlp` and create a vocabulary.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# This cell just makes sure the library paths are correct. \n",
+    "# You need to run this cell before you run the rest of this\n",
+    "# tutorial, but you can ignore the contents!\n",
+    "import os\n",
+    "import sys\n",
+    "module_path = os.path.abspath(os.path.join('../..'))\n",
+    "if module_path not in sys.path:\n",
+    "    sys.path.append(module_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from allennlp.data import Vocabulary"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Let's create an empty `Vocabulary` so we can look at the arguments it takes.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "vocab = Vocabulary(counter=None, min_count=1, max_vocab_size=100000, non_padded_namespaces=None)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "The vocabulary takes 4 arguments: \n",
+    "\n",
+    "- A counter, which is a `Dict[str, Dict[str, int]]`: This is a nested dictionary because the allennlp Vocabulary class supports the idea of \"namespaces\". A namespace is a vocabulary which is associated with a part of your data. For instance, in a sequence tagging model, you would typically have two namespaces: A namespace of words for your textual input and a namespace of tags(e.g. \"NP\", \"VP\", etc) for your labels. This counter is therefore a mapping from string namespace names to their respective mapping dictionaries of `Dict[tokens => counts]`.\n",
+    "\n",
+    "\n",
+    "- A minimum count: Tokens with smaller counts than this won't be included in your `Vocabulary`.\n",
+    "\n",
+    "\n",
+    "- A maximum vocab size: The lowest frequency words will be dropped to make your vocabulary this size.\n",
+    "\n",
+    "\n",
+    "- Non padded namespaces: These are `namespace` suffixes which won't contain padding and unknown tokens. By default, these are `*labels` and `*tags`, so any namespace you create which ends with one of these names (e.g `sequence_labels`) won't contain these additional tokens. The reason for this is explained a bit more below.\n",
```

Comment:
I would drop this argument from the example, using the default, and say above something like "We used 3 of `Vocabulary`'s arguments here:".  And then later give an example where you change it, if you want to go into that.  This also makes it easier to account for @joelgrus's recent PR.